flaky/utils.py

```diff
@@ -14,3 +14,10 @@ def ensure_unicode_string(string):
         return unicode_type(string)
     except UnicodeDecodeError:
         return string.decode('utf-8', 'replace')
+
+
+def ensure_byte_string(string):
+    try:
+        return str(string)
```

Comment:
``str`` is not the binary string type in Python 3.

See https://github.com/box/flaky/pull/46#discussion-diff-35022002 for a discussion of how to define ``binary_type``, which should be used here instead of ``str``.